climakitae/explore/warming.py

```diff
@@ -74,25 +102,29 @@ def calculate(self):
         ).squeeze()
         self.catalog_data = self.catalog_data.dropna(dim="all_sims", how="all")
         if self.wl_params.anom == "Yes":
-            gwl_times = read_csv_file(gwl_1981_2010_file, index_col=[0, 1, 2])
+            self.gwl_times = read_csv_file(gwl_1981_2010_file, index_col=[0, 1, 2])
         else:
-            gwl_times = read_csv_file(gwl_1850_1900_file, index_col=[0, 1, 2])
-        gwl_times = gwl_times.dropna(how="all")
-        self.catalog_data = clean_list(self.catalog_data, gwl_times)
-
-        self.sliced_data = self.catalog_data.groupby("all_sims").map(
-            get_sliced_data,
-            years=gwl_times,
-            window=self.wl_params.window,
-            anom=self.wl_params.anom,
-        )
-        self.sliced_data["all_sims"] = relabel_axis(self.sliced_data.all_sims)
-        self.gwl_snapshots = self.sliced_data.reduce(np.nanmean, "time")
-        self.gwl_snapshots = self.gwl_snapshots.compute()
+            self.gwl_times = read_csv_file(gwl_1850_1900_file, index_col=[0, 1, 2])
+        self.gwl_times = self.gwl_times.dropna(how="all")
+        self.catalog_data = clean_list(self.catalog_data, self.gwl_times)
+
+        self.sliced_data = {}
+        self.gwl_snapshots = {}
+        for level in self.warming_levels:
```

Comment:
This area is where the run-time significantly slows down. This changes run-time from around ~20 min. to now ~45 min., and it has to do with this for-loop being called on each of the individual warming levels. For this first iteration, I figured it would be okay to leave the code as-is ("defined as done"), and in a separate PR later, work on parallelizing this more elegantly.

Just a bit tricky with @dask.delayed calls and/or rechunking. For some reason, any calls I make with @dask.delayed don't seem to work. Happy to hop on a call to discuss ways for dealing with this too.